### ISSUE TITLE
Installation instructions

### ISSUE THREAD

**Issue by garrett** [ralphbean/fedora-hubs-prototype#1]
How does one install the proper dependencies to get this working?

(I take it `flask` is required, but should it be installed via `dnf`? Which packages should be installed? Just flask, or some additional flask-related packages? Do we want the Python 3 version? Etc.)

**Comment by ralphbean** [comment]
@garrett sorry about that.  Does d92aa19 do it for you?

**Comment by garrett** [comment]
It's kind of working... I had to start a new shell to get the `mkvirtualenv` command to work. After that, it seems like it _wants_ to work, but then I get a traceback with the DB step:

```
WARNING:fedmsg.crypto.x509:Crypto disabled ImportError('No module named M2Crypto',)
2015-05-22 10:30:05,683 INFO sqlalchemy.engine.base.Engine SELECT CAST('test plain returns' AS VARCHAR(60)) AS anon_1
INFO:sqlalchemy.engine.base.Engine:SELECT CAST('test plain returns' AS VARCHAR(60)) AS anon_1
2015-05-22 10:30:05,683 INFO sqlalchemy.engine.base.Engine ()
INFO:sqlalchemy.engine.base.Engine:()
2015-05-22 10:30:05,683 INFO sqlalchemy.engine.base.Engine SELECT CAST('test unicode returns' AS VARCHAR(60)) AS anon_1
INFO:sqlalchemy.engine.base.Engine:SELECT CAST('test unicode returns' AS VARCHAR(60)) AS anon_1
2015-05-22 10:30:05,683 INFO sqlalchemy.engine.base.Engine ()
INFO:sqlalchemy.engine.base.Engine:()
2015-05-22 10:30:05,684 INFO sqlalchemy.engine.base.Engine PRAGMA table_info("hubs")
INFO:sqlalchemy.engine.base.Engine:PRAGMA table_info("hubs")
2015-05-22 10:30:05,684 INFO sqlalchemy.engine.base.Engine ()
INFO:sqlalchemy.engine.base.Engine:()
2015-05-22 10:30:05,685 INFO sqlalchemy.engine.base.Engine PRAGMA table_info("widgets")
INFO:sqlalchemy.engine.base.Engine:PRAGMA table_info("widgets")
2015-05-22 10:30:05,685 INFO sqlalchemy.engine.base.Engine ()
INFO:sqlalchemy.engine.base.Engine:()
2015-05-22 10:30:05,685 INFO sqlalchemy.engine.base.Engine PRAGMA table_info("users")
INFO:sqlalchemy.engine.base.Engine:PRAGMA table_info("users")
2015-05-22 10:30:05,685 INFO sqlalchemy.engine.base.Engine ()
INFO:sqlalchemy.engine.base.Engine:()
2015-05-22 10:30:05,692 INFO sqlalchemy.engine.base.Engine BEGIN (implicit)
INFO:sqlalchemy.engine.base.Engine:BEGIN (implicit)
2015-05-22 10:30:05,693 INFO sqlalchemy.engine.base.Engine INSERT INTO hubs (name, summary, created_on, left_width) VALUES (?, ?, ?, ?)
INFO:sqlalchemy.engine.base.Engine:INSERT INTO hubs (name, summary, created_on, left_width) VALUES (?, ?, ?, ?)
2015-05-22 10:30:05,693 INFO sqlalchemy.engine.base.Engine ('designteam', 'The Fedora Design Team', '2015-05-22 08:30:05.693462', 8)
INFO:sqlalchemy.engine.base.Engine:('designteam', 'The Fedora Design Team', '2015-05-22 08:30:05.693462', 8)
2015-05-22 10:30:05,696 INFO sqlalchemy.engine.base.Engine ROLLBACK
INFO:sqlalchemy.engine.base.Engine:ROLLBACK
Traceback (most recent call last):
  File "populate.py", line 38, in <module>
    session.commit()
  File "/home/garrett/.virtualenvs/hubs/lib/python2.7/site-packages/sqlalchemy/orm/scoping.py", line 150, in do
    return getattr(self.registry(), name)(*args, **kwargs)
  File "/home/garrett/.virtualenvs/hubs/lib/python2.7/site-packages/sqlalchemy/orm/session.py", line 790, in commit
    self.transaction.commit()
  File "/home/garrett/.virtualenvs/hubs/lib/python2.7/site-packages/sqlalchemy/orm/session.py", line 392, in commit
    self._prepare_impl()
  File "/home/garrett/.virtualenvs/hubs/lib/python2.7/site-packages/sqlalchemy/orm/session.py", line 372, in _prepare_impl
    self.session.flush()
  File "/home/garrett/.virtualenvs/hubs/lib/python2.7/site-packages/sqlalchemy/orm/session.py", line 2004, in flush
    self._flush(objects)
  File "/home/garrett/.virtualenvs/hubs/lib/python2.7/site-packages/sqlalchemy/orm/session.py", line 2122, in _flush
    transaction.rollback(_capture_exception=True)
  File "/home/garrett/.virtualenvs/hubs/lib/python2.7/site-packages/sqlalchemy/util/langhelpers.py", line 60, in __exit__
    compat.reraise(exc_type, exc_value, exc_tb)
  File "/home/garrett/.virtualenvs/hubs/lib/python2.7/site-packages/sqlalchemy/orm/session.py", line 2086, in _flush
    flush_context.execute()
  File "/home/garrett/.virtualenvs/hubs/lib/python2.7/site-packages/sqlalchemy/orm/unitofwork.py", line 373, in execute
    rec.execute(self)
  File "/home/garrett/.virtualenvs/hubs/lib/python2.7/site-packages/sqlalchemy/orm/unitofwork.py", line 532, in execute
    uow
  File "/home/garrett/.virtualenvs/hubs/lib/python2.7/site-packages/sqlalchemy/orm/persistence.py", line 174, in save_obj
    mapper, table, insert)
  File "/home/garrett/.virtualenvs/hubs/lib/python2.7/site-packages/sqlalchemy/orm/persistence.py", line 761, in _emit_insert_statements
    execute(statement, params)
  File "/home/garrett/.virtualenvs/hubs/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 914, in execute
    return meth(self, multiparams, params)
  File "/home/garrett/.virtualenvs/hubs/lib/python2.7/site-packages/sqlalchemy/sql/elements.py", line 323, in _execute_on_connection
    return connection._execute_clauseelement(self, multiparams, params)
  File "/home/garrett/.virtualenvs/hubs/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 1010, in _execute_clauseelement
    compiled_sql, distilled_params
  File "/home/garrett/.virtualenvs/hubs/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 1078, in _execute_context
    None, None)
  File "/home/garrett/.virtualenvs/hubs/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 1339, in _handle_dbapi_exception
    exc_info
  File "/home/garrett/.virtualenvs/hubs/lib/python2.7/site-packages/sqlalchemy/util/compat.py", line 199, in raise_from_cause
    reraise(type(exception), exception, tb=exc_tb)
  File "/home/garrett/.virtualenvs/hubs/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 1073, in _execute_context
    context = constructor(dialect, self, conn, *args)
  File "/home/garrett/.virtualenvs/hubs/lib/python2.7/site-packages/sqlalchemy/engine/default.py", line 561, in _init_compiled
    self._process_executesingle_defaults()
  File "/home/garrett/.virtualenvs/hubs/lib/python2.7/site-packages/sqlalchemy/engine/default.py", line 1006, in _process_executesingle_defaults
    val = self.get_insert_default(c)
  File "/home/garrett/.virtualenvs/hubs/lib/python2.7/site-packages/sqlalchemy/engine/default.py", line 958, in get_insert_default
    return self._exec_default(column.default, column.type)
  File "/home/garrett/.virtualenvs/hubs/lib/python2.7/site-packages/sqlalchemy/engine/default.py", line 944, in _exec_default
    return default.arg(self)
  File "/home/garrett/Project/fedora/fedora-hubs-prototype/hubs/models.py", line 122, in _config_default
    print plugin.widget_arguments
sqlalchemy.exc.StatementError: (exceptions.AttributeError) 'module' object has no attribute 'widget_arguments' [SQL: u'INSERT INTO widgets (plugin, created_on, hub_id, _config, "index", "left") VALUES (?, ?, ?, ?, ?, ?)'] [parameters: [{'index': 0, 'hub_id': 'designteam', 'plugin': 'stats'}]]
```

**Comment by ralphbean** [comment]
Ah, this is a bug I introduced the other day.  It should now be fixed in bc0ef8b.  Can you try again?
